pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -456,6 +423,40 @@ func (gce *GCECloud) DeleteExternalTargetPoolAndChecks(name, region, clusterID s
 	return nil
 }
 
+// verifyUserRequestedIP checks the user-provided IP  to see whether it can be
+// used for the LB.  It also returns whether the IP is considered owned by the
+// user.
+func verifyUserRequestedIP(s CloudAddressService, region, requestedIP, fwdRuleIP, lbRef string) (ownedByUser bool, err error) {
+	if requestedIP == "" {
+		return false, nil
+	}
+	// If a specific IP address has been requested, we have to respect the
+	// user's request and use that IP. If the forwarding rule was already using
+	// a different IP, it will be harmlessly abandoned because it was only an
+	// ephemeral IP (or it was a different static IP owned by the user, in which
+	// case we shouldn't delete it anyway).
+	existingAddress, err := s.GetRegionAddressByIP(region, requestedIP)
+	if err != nil && !isNotFound(err) {
```

Comment:
If `isNotFound(err)` is true, the code would continue to check whether `requestedIP == fwdRuleIP`.  Changing this may not make the logic more readable.